### PR TITLE
[cms-v2] Add edit invoice feature

### LINKED ISSUE #1674: [cms-v2] Implement invoice edition
Add the capability of editing a submitted invoice.

### PR DESCRIPTION
**Pull request by thi4go**
This PR adds the ability to edit an invoice.

Ideally this should be merged after cmsv2 is synced with master, an open PR is up for this ~#1695~ #1705

closes #1674

### CODE REVIEW

File: src/componentsv2/InvoiceForm/InvoiceForm.jsx [hunk 1]
```diff
@@ -170,14 +164,16 @@ const InvoiceFormWrapper = ({ onSubmit, history }) => {
...
           ...others,
           month,
           year
         });
+        // Token from new invoice or from edit invoice
+        const invoiceToken = token || values.token;
         setSubmitting(false);
-        setSubmitSuccess(true);
-        history.push(`/invoices/${proposalToken}`);
+        setSubmitSuccess(true);
```
Comment (amassarwi) [inline src/componentsv2/InvoiceForm/InvoiceForm.jsx]: extra space :eyes:

File: src/componentsv2/Invoice/Invoice.jsx [hunk 2]
```diff
@@ -56,13 +60,14 @@ const Invoice = ({ invoice, extended, collapseBodyContent }) => {
   return (
     <RecordWrapper>
       {({
-        Author,
+        Author,
```
Comment (amassarwi) [inline src/componentsv2/Invoice/Invoice.jsx]: extra space :eyes:

File: src/containers/Invoice/Edit/hooks.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import { useRedux } from "src/redux";
+
+const mapStateToProps = {};
+
+const mapDispatchToProps = {
+  onEditInvoice: act.onEditInvoice
+};
+
+export function useEditInvoice(ownProps) {
+  const fromRedux = useRedux(ownProps, mapStateToProps, mapDispatchToProps);
```
Comment (amassarwi) [inline src/containers/Invoice/Edit/hooks.js]:
```suggestion
  return useRedux(ownProps, mapStateToProps, mapDispatchToProps);
```

**Comment by thi4go** [comment]
@amassarwi thanks for the review, mate. Comments addressed :+1:

File: src/containers/Invoice/Edit/Edit.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        month: invoice.input.month,
+        year: invoice.input.year
+      },
+      files: invoice.input.files,
+      lineitems: invoice.input.lineitems
+    } : null;
+
+  return (
+    <Card className="container margin-bottom-l">
+      {!loading && !!invoice ?
```
Comment (tiagoalvesdulce) [inline src/containers/Invoice/Edit/Edit.jsx]: can we create a variable with a declarative name for this boolean expression? Something like `isInvoiceLoaded`

File: src/containers/Invoice/Edit/Edit.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import get from "lodash/fp/get";
+import { withRouter } from "react-router-dom";
+import { Card } from "pi-ui";
+import { useInvoice } from "../Detail/hooks";
+import { useEditInvoice } from "./hooks";
+import { fromUSDCentsToUSDUnits } from "src/helpers";
+import InvoiceLoader from "src/componentsv2/Invoice/InvoiceLoader";
+import InvoiceForm from "src/componentsv2/InvoiceForm";
+
+const EditInvoice = ({ match }) => {
```
Comment (tiagoalvesdulce) [inline src/containers/Invoice/Edit/Edit.jsx]: Let's apply prop validation to new components. Please use propTypes for this.

File: src/pages/Invoices/Edit.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { Button, useMediaQuery } from "pi-ui";
+import MultipleContentPage from "src/componentsv2/layout/MultipleContentPage";
+import EditInvoice from "src/containers/Invoice/Edit";
+
+const PageInvoiceEdit = ({ history, match }) => {
```
Comment (tiagoalvesdulce) [inline src/pages/Invoices/Edit.jsx]: Let's apply prop validation to new components. Please use propTypes for this.

File: src/containers/Invoice/Edit/Edit.jsx (hunk 4 above)
Comment (thi4go) [reply src/containers/Invoice/Edit/Edit.jsx]: done mate :+1:

**Comment by thi4go** [comment]
Comments addressed mate @tiagoalvesdulce

**Pull request merged**